### ISSUE TITLE
Neovide Opening File Opens Same File Twice

### ISSUE THREAD

**Issue by lycheeje11y** [neovide/neovide#2609]
When I open a file like "neovide main.rs", the file opens, but it opens the file twice in two different tabs. How do I open the file only once? Am I missing something here?

**Comment by s4Dt0y** [comment]
As a side note, one workaround is to use the add the --no-tabs flag.

**Comment by fredizzimo** [comment]
I can't repeat this here, which platform are you running on?

**Comment by carlmueller** [comment]
It’s happening to me on MacOS Sonoma.

**Comment by s4Dt0y** [comment]
> I can't repeat this here, which platform are you running on?

I'm on mac sonoma as well...the context for this is trying to set up neovim with unity. I started with trying nvr, but gave up after a couple of days. So I instead tried Neovide which def works better out of the box, but opens up the files twice.

**Comment by carlmueller** [comment]
The following workaround works for me.

`open -a neovide filename`

**Comment by fredizzimo** [comment]
Very strange, maybe the commandline parser interprets the file as both a file to open and neovim arguments. Can you try with `neovide main.rs --`, forcing the neovim arguments to be empty?

Although, it should be the same on all platforms.

**Comment by carlmueller** [comment]
`neovide test.txt --` still opens the file twice.

**Comment by carlmueller** [comment]
`neovide test.txt --log` does open the file twice (while `neovide --log test.txt` does not).  I'm attaching the log file.
[neovide_rCURRENT.log](https://github.com/user-attachments/files/15568416/neovide_rCURRENT.log)

**Comment by fredizzimo** [comment]
Here's the stupid design Apple has decided to implement
https://lists.apple.com/archives/cocoa-dev/2009/May/msg00480.html,

TLDR; there's no way to disable the built in parsing of the arguments, so we need to ignore the openfile requests for them, or maybe always only use openfile, and ignore the initial arguments. 

The problem is that I can't find any official documentation of this, and nothing about what the scheme it uses, if everything that looks like a file is passed to `openFile`, or if the file actually has to exist for example. And in both cases we need to be able to filter the list of arguments.

I guess that's something for the one actually fixing the bug to find out...

**Comment by falcucci** [comment]
For ppl who wants this fixed the workaround would be disabling neovide `tabs` to make it appears as Buffer only using
```bash
neovide --no-tabs <file>
```

or even at our [config.toml](https://neovide.dev/config-file.html#available-settings) file.

```toml
tabs = false
```

**Comment by fredizzimo** [comment]
I'm re-opening this, as I don't think no-tabs is the correct long-term fix.

**Comment by alexventuraio** [comment]
I'm still facing this same issue when opening a folder with `neovide .` in the lates `0.13.2` version in MacOS Sonoma 14.5 (23F79)

**Comment by ernie** [comment]
It seems like this multiple-tab thing is happening specifically depending on the number of args passed in.

`neovide blah` => two tabs
`neovide --fork blah` => one tab
`neovide -- blah` => one tab
`neovide blah1 blah2` => four tabs: blah1, blah1, blah2, blah2
`neovide -- blah1 blah2` => three tabs: blah1, blah2, blah2
`neovide -- blah1 blah2 blah3` => five tabs: blah1, blah2, blah3, blah2, blah3

**Comment by brnt** [comment]
Just upgraded from 0.12.2, which doesn't have the double-tab issue (from terminal or finder). The `--no-tabs` trick works to suppress the second tab when opening neovide 0.13.3 from the terminal, but (obviously) doesn't change the behavior from the finder (still double-tabbed).

One other change—perhaps unrelated: opening version 0.13.3 from the terminal no longer pushes the application to the background, which the previous install did.

**Comment by viotti** [comment]
If you use `neovide --no-tabs`, you can then run `:tab sball` to restore the tabs.

**Comment by fboundp** [comment]
> TLDR; there's no way to disable the built in parsing of the arguments, so we need to ignore the openfile requests for them, or maybe always only use openfile, and ignore the initial arguments.

The AppKit parsing of arguments can be disabled by setting the undocumented `NSTreatUnknownArgumentsAsOpen` user defaults option to false. PR #2798 does so.
